build.gradle

```diff
@@ -55,40 +35,56 @@ license {
 
 java {
     withSourcesJar()
+    withJavadocJar()
 }
 
-//This duplicates the publish for some reason
-gradle.taskGraph.whenReady { graph -> publishMavenJavaPublicationToMavenRepository.onlyIf { false } }
 publishing {
     publications {
-        mavenJava(MavenPublication) {
-            repositories {
-                maven {
-                    if (project.hasProperty('forgeMavenPassword')) {
-                        credentials {
-                            username project.properties.forgeMavenUser
-                            password project.properties.forgeMavenPassword
-                        }
-                        url 'https://files.minecraftforge.net/maven/manage/upload'
-                    } else {
-                        url 'file://' + rootProject.file('repo').getAbsolutePath()
-                    }
+        withType(MavenPublication) {
+            pom {
+                description = project.description
+            }
+        }
+    }
+    repositories {
+        maven {
+            if (project.hasProperty('forgeMavenPassword')) {
+                credentials {
+                    username project.properties.forgeMavenUser
+                    password project.properties.forgeMavenPassword
                 }
+                url 'https://files.minecraftforge.net/maven/manage/upload'
+            } else {
+                url 'file://' + rootProject.file('repo').getAbsolutePath()
             }
```

Comment:
I saw there was something for this [in GradleUtils](https://github.com/MinecraftForge/GradleUtils/blob/6a1de99ddc5cba65e4fcec3c9a9efd8f26b63e5b/src/main/groovy/net/minecraftforge/gradleutils/GradleUtilsExtension.groovy#L130), but couldn't see it used peeking at a few of the projects you've updated. Does it work, and if so would it be preferred here?

Reply:
It does work, but no projects use it at the moment. I'm not sure if it would be preferred, that's up to lex. I added it with the intent of being able to update the maven publishing process without hassle on forge projects. However in hindsight, you still have to declare the forge maven in the first place to get GradleUtils, and the Jenkinsfile is still hardcoded to provide the specific environment variables it does. So until that part of the Jenkinsfile is abstracted out to forge-shared-library, if ever, it's basically useless to use the closure provided by GradleUtils.